🐛 Bug:  At least one dependency name should be specified for removal after --mode initialization

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

After fixing #722, we should pass with no errors.

### Actual

```plaintext
Error: Failed uninstalling initialization-only packages
    at withSpinner (file:///Users/josh/repos/template-typescript-node-package/lib/shared/cli/spinners.js:16:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async initializeWithOptions (file:///Users/josh/repos/template-typescript-node-package/lib/initialize/initializeWithOptions.js:47:5)
    ... 2 lines matching cause stack trace ...
    at async Object.initialize (file:///Users/josh/repos/template-typescript-node-package/lib/initialize/index.js:9:10)
    at async bin (file:///Users/josh/repos/template-typescript-node-package/lib/bin/index.js:47:10)
    at async file:///Users/josh/repos/template-typescript-node-package/bin/index.js:4:20 {
  [cause]: Error: Command failed with exit code 1: pnpm remove
   ERR_PNPM_MUST_REMOVE_SOMETHING  At least one dependency name should be specified for removal
      at makeError (file:///Users/josh/repos/template-typescript-node-package/node_modules/.pnpm/execa@8.0.1/node_modules/execa/lib/error.js:60:11)
      at handlePromise (file:///Users/josh/repos/template-typescript-node-package/node_modules/.pnpm/execa@8.0.1/node_modules/execa/index.js:124:26)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async removeDependencies (file:///Users/josh/repos/template-typescript-node-package/lib/shared/packages.js:9:3)
      at async uninstallPackages (file:///Users/josh/repos/template-typescript-node-package/lib/steps/uninstallPackages.js:9:3)
      at async withSpinner (file:///Users/josh/repos/template-typescript-node-package/lib/shared/cli/spinners.js:11:20)
      at async initializeWithOptions (file:///Users/josh/repos/template-typescript-node-package/lib/initialize/initializeWithOptions.js:47:5)
      at async run (file:///Users/josh/repos/template-typescript-node-package/lib/initialize/index.js:11:7)
      at async runOrRestore (file:///Users/josh/repos/template-typescript-node-package/lib/shared/runOrRestore.js:7:5)
      at async Object.initialize (file:///Users/josh/repos/template-typescript-node-package/lib/initialize/index.js:9:10) {
    shortMessage: 'Command failed with exit code 1: pnpm remove',
    command: 'pnpm remove',
    escapedCommand: 'pnpm remove',
    exitCode: 1,
    signal: undefined,
    signalDescription: undefined,
    stdout: ' ERR_PNPM_MUST_REMOVE_SOMETHING  At least one dependency name should be specified for removal',
    stderr: '',
    cwd: '/Users/josh/repos/temp/testing-cta-6',
    failed: true,
    timedOut: false,
    isCanceled: false,
    killed: false
  }
}
```



### Additional Info

😂 it never ends...

--- a/src/shared/packages.test.ts
+++ b/src/shared/packages.test.ts
@@ -2,32 +2,39 @@ import { describe, expect, it, vi } from "vitest";
 
 import { removeDependencies } from "./packages.js";
 
-const mock$ = vi.fn();
+const mockExecaCommand = vi.fn();
 
 vi.mock("execa", () => ({
-	get $() {
-		return mock$;
+	get execaCommand() {
+		return mockExecaCommand;
 	},
 }));
 
 describe("removeDependencies", () => {
+	it("removes all packages that already exist when all already exist", async () => {
+		await removeDependencies(["one", "two"], {
+			one: "1.2.3",
+			two: "4.5.6",
+		});
+
+		expect(mockExecaCommand.mock.calls).toMatchInlineSnapshot(`
+			[
+			  [
+			    "pnpm remove one two",
+			  ],
+			]
+		`);
+	});
+
 	it("removes only packages that already exist when some don't exist", async () => {
 		await removeDependencies(["exists", "missing"], {
 			exists: "1.2.3",
 		});
 
-		expect(mock$.mock.calls).toMatchInlineSnapshot(`
+		expect(mockExecaCommand.mock.calls).toMatchInlineSnapshot(`
 			[
 			  [
-			    [
-			      "pnpm remove ",
-			      "",
-			      "",
-			    ],
-			    [
-			      "exists",
-			    ],
-			    "",
+			    "pnpm remove exists",
 			  ],
 			]
 		`);
@@ -42,18 +49,10 @@ describe("removeDependencies", () => {
 			"-D",
 		);
 
-		expect(mock$.mock.calls).toMatchInlineSnapshot(`
+		expect(mockExecaCommand.mock.calls).toMatchInlineSnapshot(`
 			[
 			  [
-			    [
-			      "pnpm remove ",
-			      "",
-			      "",
-			    ],
-			    [
-			      "exists",
-			    ],
-			    " -D",
+			    "pnpm remove exists -D",
 			  ],
 			]
 		`);
@@ -62,6 +61,6 @@ describe("removeDependencies", () => {
 	it("does nothing when no packages already exist", async () => {
 		await removeDependencies(["missing"]);
 
-		expect(mock$.mock.calls).toMatchInlineSnapshot("[]");
+		expect(mockExecaCommand.mock.calls).toMatchInlineSnapshot("[]");
 	});
 });

--- a/src/shared/packages.test.ts
+++ b/src/shared/packages.test.ts
@@ -29,7 +29,7 @@ describe("removeDependencies", () => {
 		`);
 	});
 
-	it("adds a flag removes packages when one is provided", async () => {
+	it("adds a flag to removing packages when one is provided", async () => {
 		await removeDependencies(
 			["exists", "missing"],
 			{

--- a/src/shared/packages.test.ts
+++ b/src/shared/packages.test.ts
@@ -21,11 +21,9 @@ describe("removeDependencies", () => {
 			  [
 			    [
 			      "pnpm remove ",
-			      " ",
 			      "",
 			    ],
 			    "exists",
-			    "",
 			  ],
 			]
 		`);
@@ -45,11 +43,9 @@ describe("removeDependencies", () => {
 			  [
 			    [
 			      "pnpm remove ",
-			      " ",
 			      "",
 			    ],
-			    "exists",
-			    "-D",
+			    "exists -D",
 			  ],
 			]
 		`);

--- a/src/shared/packages.test.ts
+++ b/src/shared/packages.test.ts
@@ -22,8 +22,12 @@ describe("removeDependencies", () => {
 			    [
 			      "pnpm remove ",
 			      "",
+			      "",
 			    ],
-			    "exists",
+			    [
+			      "exists",
+			    ],
+			    "",
 			  ],
 			]
 		`);
@@ -44,8 +48,12 @@ describe("removeDependencies", () => {
 			    [
 			      "pnpm remove ",
 			      "",
+			      "",
 			    ],
-			    "exists -D",
+			    [
+			      "exists",
+			    ],
+			    " -D",
 			  ],
 			]
 		`);

--- a/src/shared/packages.test.ts
+++ b/src/shared/packages.test.ts
@@ -1,0 +1,63 @@
+import { describe, expect, it, vi } from "vitest";
+
+import { removeDependencies } from "./packages.js";
+
+const mock$ = vi.fn();
+
+vi.mock("execa", () => ({
+	get $() {
+		return mock$;
+	},
+}));
+
+describe("removeDependencies", () => {
+	it("removes only packages that already exist when some don't exist", async () => {
+		await removeDependencies(["exists", "missing"], {
+			exists: "1.2.3",
+		});
+
+		expect(mock$.mock.calls).toMatchInlineSnapshot(`
+			[
+			  [
+			    [
+			      "pnpm remove ",
+			      " ",
+			      "",
+			    ],
+			    "exists",
+			    "",
+			  ],
+			]
+		`);
+	});
+
+	it("adds a flag removes packages when one is provided", async () => {
+		await removeDependencies(
+			["exists", "missing"],
+			{
+				exists: "1.2.3",
+			},
+			"-D",
+		);
+
+		expect(mock$.mock.calls).toMatchInlineSnapshot(`
+			[
+			  [
+			    [
+			      "pnpm remove ",
+			      " ",
+			      "",
+			    ],
+			    "exists",
+			    "-D",
+			  ],
+			]
+		`);
+	});
+
+	it("does nothing when no packages already exist", async () => {
+		await removeDependencies(["missing"]);
+
+		expect(mock$.mock.calls).toMatchInlineSnapshot("[]");
+	});
+});

--- a/src/shared/packages.ts
+++ b/src/shared/packages.ts
@@ -1,4 +1,4 @@
-import { $ } from "execa";
+import { execaCommand } from "execa";
 
 import { readFileSafe } from "./readFileSafe.js";
 import { PartialPackageData } from "./types.js";
@@ -17,6 +17,8 @@ export async function removeDependencies(
 	const present = packageNames.filter((packageName) => packageName in existing);
 
 	if (present.length) {
-		await $`pnpm remove ${present}${flags ? ` ${flags}` : ""}`;
+		await execaCommand(
+			`pnpm remove ${present.join(" ")}${flags ? ` ${flags}` : ""}`,
+		);
 	}
 }

--- a/src/shared/packages.ts
+++ b/src/shared/packages.ts
@@ -17,6 +17,6 @@ export async function removeDependencies(
 	const present = packageNames.filter((packageName) => packageName in existing);
 
 	if (present.length) {
-		await $`pnpm remove ${present.join(" ")} ${flags}`;
+		await $`pnpm remove ${[...present, flags].filter(Boolean).join(" ")}`;
 	}
 }

--- a/src/shared/packages.ts
+++ b/src/shared/packages.ts
@@ -11,22 +11,12 @@ export async function readPackageData() {
 
 export async function removeDependencies(
 	packageNames: string[],
-	packageData: PartialPackageData,
+	existing: Record<string, string> = {},
+	flags = "",
 ) {
-	await $`pnpm remove ${packageNames.filter(
-		packageExists(packageData.dependencies),
-	)}`;
-}
+	const present = packageNames.filter((packageName) => packageName in existing);
 
-export async function removeDevDependencies(
-	packageNames: string[],
-	packageData: PartialPackageData,
-) {
-	await $`pnpm remove ${packageNames.filter(
-		packageExists(packageData.devDependencies),
-	)} -D`;
-}
-
-function packageExists(listing: Record<string, string> = {}) {
-	return (packageName: string) => packageName in listing;
+	if (present.length) {
+		await $`pnpm remove ${present.join(" ")} ${flags}`;
+	}
 }

--- a/src/shared/packages.ts
+++ b/src/shared/packages.ts
@@ -17,6 +17,6 @@ export async function removeDependencies(
 	const present = packageNames.filter((packageName) => packageName in existing);
 
 	if (present.length) {
-		await $`pnpm remove ${[...present, flags].filter(Boolean).join(" ")}`;
+		await $`pnpm remove ${present}${flags ? ` ${flags}` : ""}`;
 	}
 }

--- a/src/steps/finalizeDependencies.test.ts
+++ b/src/steps/finalizeDependencies.test.ts
@@ -11,13 +11,9 @@ vi.mock("execa", () => ({
 	},
 }));
 
-const mockRemoveDevDependencies = vi.fn();
-
 vi.mock("../shared/packages.js", () => ({
 	readPackageData: () => [],
-	get removeDevDependencies() {
-		return mockRemoveDevDependencies;
-	},
+	removeDependencies: vi.fn(),
 }));
 
 const options = {

--- a/src/steps/finalizeDependencies.ts
+++ b/src/steps/finalizeDependencies.ts
@@ -1,6 +1,6 @@
 import { execaCommand } from "execa";
 
-import { readPackageData, removeDevDependencies } from "../shared/packages.js";
+import { readPackageData, removeDependencies } from "../shared/packages.js";
 import { Options } from "../shared/types.js";
 
 export async function finalizeDependencies(options: Options) {
@@ -68,9 +68,10 @@ export async function finalizeDependencies(options: Options) {
 
 	if (!options.excludeContributors) {
 		await execaCommand(`npx all-contributors-cli generate`);
-		await removeDevDependencies(
+		await removeDependencies(
 			["all-contributors-cli", "all-contributors-for-repository"],
-			await readPackageData(),
+			(await readPackageData()).devDependencies,
+			"-D",
 		);
 	}
 

--- a/src/steps/uninstallPackages.ts
+++ b/src/steps/uninstallPackages.ts
@@ -1,10 +1,6 @@
 import { $ } from "execa";
 
-import {
-	readPackageData,
-	removeDependencies,
-	removeDevDependencies,
-} from "../shared/packages.js";
+import { readPackageData, removeDependencies } from "../shared/packages.js";
 
 export async function uninstallPackages() {
 	const packageData = await readPackageData();
@@ -24,10 +20,10 @@ export async function uninstallPackages() {
 			"replace-in-file",
 			"title-case",
 		],
-		packageData,
+		packageData.dependencies,
 	);
 
-	await removeDevDependencies(
+	await removeDependencies(
 		[
 			"@octokit/request-error",
 			"@types/git-url-parse",
@@ -38,7 +34,8 @@ export async function uninstallPackages() {
 			"globby",
 			"tsx",
 		],
-		packageData,
+		packageData.devDependencies,
+		"-D",
 	);
 
 	await $`pnpm add prettier -D`;

